scripts/provision_layers.py

```diff
@@ -0,0 +1,73 @@
+from argparse import ArgumentParser
+import logging
+from pathlib import Path
+import subprocess
+import sys
+from zipfile import ZipFile
+
+import boto3
+
+from azul import config
+from azul.logging import configure_script_logging
+
+log = logging.getLogger(__name__)
+
+
+class LayersProvisioner:
+    layer_dir = Path(config.project_root) / 'lambdas' / 'layer'
+    out_dir = layer_dir / '.chalice' / 'terraform'
+
+    def __init__(self):
+        self.client = boto3.client('s3')
+
+    def provision_layer(self, bucket, key):
+        log.info('Provisioning layer in bucket: %s with key %s', bucket, key)
+        if self.update_required(bucket, key):
+            log.info('Updating layer.zip')
+            self.update_layer(bucket, key)
+        else:
+            log.info('No update necessary')
+
+    def update_required(self, bucket, key) -> bool:
+        log.info('Checking if layer.zip needs to be updated')
+        try:
+            # Since the object key is content-addressable, just checking for
+            # the file's presence is sufficient
+            self.client.head_object(Bucket=bucket, Key=key)
+        except self.client.exceptions.ClientError as e:
+            if e.response['Error']['Code'] == '404':
+                return True
+        else:
+            return False
+
+    def update_layer(self, bucket, key):
+        input_zip = self.out_dir / 'deployment.zip'
+        layer_zip = self.out_dir / 'layer.zip'
+        self.build_layer_zip(layer_zip, input_zip)
+        self.client.upload_file(str(layer_zip), bucket, key)
```

Comment:
does terraform handle the layer registration? there might need to be some layer publishing. https://docs.aws.amazon.com/lambda/latest/dg/configuration-layers.html#configuration-layers-manage

Reply:
Yes, Terraform handles registering the layer. The Terraform configuration for that is here https://github.com/DataBiosphere/azul/pull/1655/files#diff-6d607bdc1bf3a3126d91f66c2c027df1R1